src/finufft.cpp

```diff
@@ -1092,6 +1073,13 @@ int FINUFFT_EXECUTE(FINUFFT_PLAN p, CPX *cj, CPX *fk) {
   CNTime timer;
   timer.start();
 
+#ifdef FINUFFT_USE_DUCC0
+  std::vector<CPX> fwBatch_(p->nf * p->batchSize); // the big workspace
```

Comment:
See above comment - could you explain why this isn't better done in the plan stage?

Reply:
The motivation is to reduce memory consumption of plans. I remember the HVOX people complaining at some point that finufft plans are huge, and with FFTW this is not so easy to avoid. With DUCC it's easier and I think it's worth the slight performance penalty.

Reply:
I think, I guess we could always use a caching allocator to avoid the issue. I'd rather keep the allocation in the plan for now and maybe switch to this with a flag or something. I do prefer having the same behavior for now.

Reply:
If you are really sure ... for me this was one of the two big selling points of the patch (the other being not waiting for the FFTW planner).

Reply:
It's reverted now ... but I sincerely hope that you will change your mind again! :-)

Reply:
Ok, interesting that this was a selling point. Let's think about where this big malloc is done in the future, as a separate issue. We would like to understand how HVOX people are using the plan interface. The big malloc has to be done *somewhere* because the FFT is about to write into it!   But if one were to keep multiple plans around that were not going to be done simultaneously, then, yes, it could be wasteful of RAM - but that seems like a weird use-case. Maybe people are using FINUFFT in a way I hadn't anticipated.

Reply:
A very common situation would be iterative approximate inversion of a NUFFT via conjugate gradient or similar. For that you need (I think) two FINUFFT plans that are used alternatingly, and therefore there is always an overhead of a fine grid (and a copy of the coordinates of nonuniform points, but that's a different topic again).

Reply:
Thank you very much for the clarification! I'm definitely looking forward to reading about the improved algorithm!